### PR TITLE
Fix bazel BUILD rule and suppress pkg-config errors

### LINKED ISSUE #79499: hack/update-bazel produces warnings
**What happened**:
Got the following error while trying to run `hack/update-bazel.sh`:
```
gazelle: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: error reading go file: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: pkg-config not supported: #cgo pkg-config: libseccomp
gazelle: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: error reading go file: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: pkg-config not supported: #cgo pkg-config: libseccomp
``` 

**How to reproduce it (as minimally and precisely as possible)**:
Ran this on a fresh install of ubuntu 19.04, Go 1.12.6, and bazel 0.26.1.

**Anything else we need to know?**:
Seems like the change might have been due to the update in vendored deps from #78187 ( https://github.com/kubernetes/kubernetes/pull/78187#discussion_r294546736 ).
Although this script is run in CI where it is working so may be some set of recommende pkg versions would help prevent this problem?

**Environment**:
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`): ubuntu 19.04
- Kernel (e.g. `uname -a`): 5.0.0-15-generic
- bazel: 0.26.1
- Go: 1.12.6

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix bazel BUILD rule and suppresses the following gazelle output in hack/update-bazel.sh:
```
gazelle: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/google/cadvisor/nvm/machine_libipmctl.go: error reading go file: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/google/cadvisor/nvm/machine_libipmctl.go: pkg-config not supported: #cgo pkg-config: libipmctl
gazelle: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: error reading go file: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: pkg-config not supported: #cgo pkg-config: libseccomp
gazelle: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: error reading go file: /home/user/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: pkg-config not supported: #cgo pkg-config: libseccomp
```

```sh
# pkg-config --cflags --libs libipmctl
 -lipmctl  
```

**Which issue(s) this PR fixes**:

Fixes #79499

Ref: #79502

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
We should not suppress any errors we aren't working around by manually configuring the equivilant settings.

**Comment by SataQiu** [comment]
> We should not suppress any errors we aren't working around by manually configuring the equivilant settings.

Hi @BenTheElder Do you find a better way to handle these errors? 
I just saw https://github.com/kubernetes/kubernetes/issues/79499#issuecomment-614486261.
If there is a better way, we should adopt it. :smile:

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/79287 is an example of actually fixing it before suppressing. It has to be done manually by inspection.

File: vendor/github.com/google/cadvisor/nvm/BUILD [hunk 1]
```diff
@@ -4,6 +4,7 @@ go_library(
     name = "go_default_library",
     srcs = ["machine_no_libipmctl.go"],
     cgo = True,
+    clinkopts = select({"@io_bazel_rules_go//go/platform:linux":["-lipmctl",],"//conditions:default":[],}),
```
[the remark this answers is not in the document]
Comment (SataQiu) [reply vendor/github.com/google/cadvisor/nvm/BUILD:7]: Emm... Is there a good way to dynamically determine link with libmpctl or not? A bit tricky :sweat_smile:

[the remark this answers is not in the document]
Comment (iwankgb) [reply vendor/github.com/google/cadvisor/nvm/BUILD:7]: cAdvisor can be built without `libipmctl` and it is default behaviour. To build *with* the lib you need to specify a build tag.

[the remark this answers is not in the document]
Comment (BenTheElder) [reply vendor/github.com/google/cadvisor/nvm/BUILD:7]: maybe we should not set this then and just suppress the pkg-config error.

[the remark this answers is not in the document]
Comment (SataQiu) [reply vendor/github.com/google/cadvisor/nvm/BUILD]: Agree, it is better to keep the default behavior as before~

File: hack/update-bazel.sh [hunk 2]
```diff
@@ -64,12 +64,16 @@ done
 # filter out known pkg-config error (see buildozer workaround below)
 # NOTE: the `|| exit "${PIPESTATUS[0]}"` is to ignore grep errors 
 # while preserving the exit code of gazelle
+KNOWN_PKG_CONFIG_ERROR_PACKAGES=(
```
Comment (BenTheElder) [inline hack/update-bazel.sh:67]: we should add a warning to be careful when ignoring these. in the case of libseccomp we actually needed to modify the script below to handle it

**Comment by iwankgb** [comment]
Besides `libipmctl` you might encounter similar issues with `libpfm4`. Resolution is pretty similar: default build does not need it.

File: hack/update-bazel.sh (hunk 2 above)
Comment (SataQiu) [reply hack/update-bazel.sh:67]: How about adding some comments like this? @BenTheElder 
```
Warning: Please be careful to ignore these pkg-config errors. 
As gazelle doesn't support pkg-config, if we suppress these errors here, we actually need to modify the script below to handle it manually.
For instance, in the case of `libseccomp`, we set the link option `-lseccomp` to make it work properly.
```

Comment (BenTheElder) [reply hack/update-bazel.sh:67]: > Warning: Please be careful to ignore these pkg-config errors. 

I think this should read:
> Warning: Please be careful when ignoring these pkg-config errors.

Comment (BenTheElder) [reply hack/update-bazel.sh:67]: thanks!

Comment (SataQiu) [reply hack/update-bazel.sh]: @BenTheElder 
I have updated the code. Thanks for your help! :)

**Comment by SataQiu** [comment]
/assign @fejta 
I think this PR can be LGTM now 😃

**Pull request merged**